envoy/upstream/upstream.h

```diff
@@ -713,7 +718,7 @@ class TypedLoadBalancerFactory;
 /**
  * Information about a given upstream cluster.
```

Comment:
It might be nice to add a comment which alludes to the fact that this is also a factor for creating upstream filters?